client/analytics/settings/historical-data/index.js

```diff
@@ -0,0 +1,185 @@
+			customersTotal,
+			ordersProgress,
+			ordersTotal,
+			period,
+			skipChecked,
+			step,
+		} = this.state;
+		const isInProgress = this.isInProgress();
+
+		return (
+			<Fragment>
+				<div className="woocommerce-setting">
+					<div className="woocommerce-setting__label" id="import-historical-data-label">
+						{ __( 'Import Historical Data:', 'woocommerce-admin' ) }
+					</div>
+					<div className="woocommerce-setting__input">
+						<span className="woocommerce-setting__help">
+							{ __(
+								'This tool populates historical analytics data by processing customers ' +
+									'and orders created prior to activating WooCommerce Admin.',
+								'woocommerce-admin'
+							) }
+						</span>
+						{ step !== 'finished' && (
+							<Fragment>
+								<HistoricalDataPeriodSelector
+									dateFormat={ this.dateFormat }
+									disabled={ isInProgress }
+									onPeriodChange={ this.onPeriodChange }
+									onDateChange={ this.onDateChange }
+									value={ period }
+								/>
+								<HistoricalDataSkipCheckbox
+									disabled={ isInProgress }
+									checked={ skipChecked }
+									onChange={ this.onSkipChange }
+								/>
+								<HistoricalDataProgress
+									label={ __( 'Registered Customers', 'woocommerce-admin' ) }
+									progress={ customersProgress }
+									total={ customersTotal }
+								/>
+								<HistoricalDataProgress
+									label={ __( 'Orders', 'woocommerce-admin' ) }
+									progress={ ordersProgress }
+									total={ ordersTotal }
+								/>
+							</Fragment>
+						) }
+						<HistoricalDataStatus status={ step } />
+					</div>
+				</div>
+				<div className="woocommerce-settings__actions">
+					{ step === 'ready' && (
+						<Button isPrimary onClick={ this.changeStep( 'customers' ) }>
+							{ __( 'Start', 'woocommerce-admin' ) }
+						</Button>
+					) }
+					{ isInProgress && (
+						<Fragment>
+							<Button
+								className="woocommerce-settings-historical-data__action-button"
+								isPrimary
+								onClick={ this.changeStep( 'stopped' ) }
+							>
+								{ __( 'Stop Import', 'woocommerce-admin' ) }
+							</Button>
+							<div className="woocommerce-setting__help woocommerce-settings-historical-data__action-help">
+								{ __(
+									'Imported data will not be lost if the import is stopped.',
+									'woocommerce-admin'
+								) }
+								<br />
+								{ __(
+									'Navigating away from this page will not affect the import.',
+									'woocommerce-admin'
+								) }
+							</div>
+						</Fragment>
+					) }
+					{ step === 'stopped' && (
+						<Fragment>
+							<Button isPrimary onClick={ this.changeStep( 'customers' ) }>
+								{ __( 'Start', 'woocommerce-admin' ) }
+							</Button>
+							<Button isDefault onClick={ this.changeStep( 'finished' ) }>
+								{ __( 'Delete Previously Imported Data', 'woocommerce-admin' ) }
+							</Button>
+						</Fragment>
+					) }
+					{ step === 'finished' && (
```

Comment:
I think instead of using steps here we can let the data from the API drive the available options.  Steps might also create issues where data was imported, but it's not possible to delete it until running through all the steps here again.
 
Would `inProgress` and `hasImportedData` be enough to remove all the step logic?

Reply:
I re-implemented it in its own component trying to make the logic simpler and getting rid of `step`. Let me know how it looks!